zipkin-collector/pulsar/src/main/java/zipkin2/collector/pulsar/PulsarSpanConsumer.java

```diff
@@ -0,0 +1,85 @@
+/*
+ * Copyright The OpenZipkin Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+package zipkin2.collector.pulsar;
+
+import org.apache.pulsar.client.api.Consumer;
+import org.apache.pulsar.client.api.Message;
+import org.apache.pulsar.client.api.MessageListener;
+import org.apache.pulsar.client.api.PulsarClient;
+import org.apache.pulsar.client.api.PulsarClientException;
+import org.apache.pulsar.client.api.SubscriptionType;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import zipkin2.Callback;
+import zipkin2.collector.Collector;
+import zipkin2.collector.CollectorMetrics;
+
+import java.io.Closeable;
+import java.util.Map;
+
+public class PulsarSpanConsumer implements Closeable {
+  static final Callback<Void> NOOP = new Callback<>() {
+    @Override public void onSuccess(Void value) {
+    }
+
+    @Override public void onError(Throwable t) {
+    }
+  };
+
+  private static final Logger LOG = LoggerFactory.getLogger(PulsarSpanConsumer.class);
+  private final String topic;
+  private final Map<String, Object> consumerProps;
+  private final PulsarClient client;
+  private final Collector collector;
+  private final CollectorMetrics metrics;
+  private Consumer<byte[]> consumer;
+
+  public PulsarSpanConsumer(String topic, Map<String, Object> consumerProps, PulsarClient client, Collector collector, CollectorMetrics metrics) {
+    this.topic = topic;
+    this.consumerProps = consumerProps;
+    this.client = client;
+    this.collector = collector;
+    this.metrics = metrics;
+  }
+
+  public void startConsumer() throws PulsarClientException {
+    consumer = client.newConsumer()
+        .topic(topic)
+        .subscriptionType(SubscriptionType.Shared)
+        .loadConf(consumerProps)
+        .messageListener(new ZipkinMessageListener<>(collector, metrics))
+        .subscribe();
+  }
+
+  @Override public void close() {
+    try {
+      if (consumer != null) {
+        consumer.close();
```

Comment:
Sorry, I forgot to add the `null` assignment to the consumer:
```suggestion
        consumer.close();
        consumer = null;
```